torch/csrc/jit/codegen/cuda/kernel_cache.h

```diff
@@ -56,10 +56,34 @@ class TORCH_CUDA_CU_API FusionKernelRuntime {
     }
   }
 
+  //! query if we already have a compiled kernel for execution
+  bool isCompiled() {
+    std::unique_lock<std::mutex> lock0(mutex_, std::try_to_lock);
+    std::unique_lock<std::mutex> lock1(compiling_, std::try_to_lock);
+    if (!lock0.owns_lock() || !lock0.owns_lock()) {
```

Comment:
nit: typo?

Reply:
Also would you need a lock to check this? Compiled can only go from `false` to `true`.

Reply:
Oops. good catch~~

This part needs to be thought out later. I don't think what I have here is right. <- I should try_to_lock on compiling_ and lock mutex_.
Two separate mutex is needed because: 1. we want to safe iterating on executors_ while no other threads writing to it. -> hence the need of mutex_; 2. we don't want to duplicate compilation of the same kernel. So if a compiling_ is going on, we might not want to compile another kernel.

I need to go back and double check our heuristic search before refactor this. But the typo is real and I'll fix it.